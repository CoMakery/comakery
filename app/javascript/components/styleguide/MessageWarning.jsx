import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Message from './Message'

class MessageWarning extends React.Component {
  render() {
    const {className, ...other} = this.props

    const classnames = classNames(
      'message__warning',
      className
    )

    return (
      <React.Fragment>
        <Message
          className={classnames} {...other}
          severity="warning"
        />
      </React.Fragment>
    )
  }
}

MessageWarning.propTypes = {
  className: PropTypes.string,
  text     : PropTypes.string
}
MessageWarning.defaultProps = {
  className: '',
  text     : 'warning text'
}
export default MessageWarning
