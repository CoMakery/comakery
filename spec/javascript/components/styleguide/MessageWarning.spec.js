import React from 'react'
import { shallow, mount, render } from 'enzyme'
import MessageWarning from 'components/styleguide/MessageWarning'

describe('MessageWarning', () => {
    it('renders correctly without props', () => {
      const wrapper = shallow(<MessageWarning/>)

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.exists('.message__warning')).toBe(true)
    })

    it('renders correctly with custom className', () => {
      const wrapper = shallow(<MessageWarning className='__test' />)

      expect(wrapper.exists('.message__warning.__test')).toBe(true)
    })
})
