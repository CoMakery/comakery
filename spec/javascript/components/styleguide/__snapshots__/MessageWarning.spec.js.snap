// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageWarning renders correctly without props 1`] = `
<Fragment>
  <Message
    className="message__warning"
    severity="warning"
    text="warning text"
  />
</Fragment>
`;
